src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -0,0 +1,139 @@
+                    incomingMessage.Add(buffer.Array);
+                    totalBytes += receiveResult.Count;
+                } while (!receiveResult.EndOfMessage);
+
+                Message message;
+                if (incomingMessage.Count > 1)
+                {
+                    var messageBuffer = new byte[totalBytes];
+                    var offset = 0;
+                    for (var i = 0 ; i < incomingMessage.Count; i++)
+                    {
+                        Buffer.BlockCopy(incomingMessage[i], 0, messageBuffer, offset, incomingMessage[i].Length);
+                        offset += incomingMessage[i].Length;
+                    }
+                    
+                    message = new Message(ReadableBuffer.Create(messageBuffer).Preserve(), receiveResult.MessageType == 0 ? Format.Text : Format.Binary, receiveResult.EndOfMessage);
+                }
+                else
+                {
+                    message = new Message(ReadableBuffer.Create(incomingMessage[0]).Preserve(), receiveResult.MessageType == 0 ? Format.Text : Format.Binary, receiveResult.EndOfMessage);
+                }
+
+                while (await _application.Output.WaitToWriteAsync(cancellationToken))
+                {
+                    if (_application.Output.TryWrite(message))
+                    {
+                        incomingMessage.Clear();
+                        break;
+                    }
+                }
+            }
+        }
+
+        private async Task SendMessages(Uri sendUrl, CancellationToken cancellationToken)
+        {
+            while (await _application.Input.WaitToReadAsync(cancellationToken))
+            {
+                Message message;
+                while (!cancellationToken.IsCancellationRequested && _application.Input.TryRead(out message))
+                {
+                    using (message)
+                    {
+                        try
+                        {
+                            await _webSocket.SendAsync(new ArraySegment<byte>(message.Payload.Buffer.ToArray()),
+                            message.MessageFormat == Format.Text ? WebSocketMessageType.Text : WebSocketMessageType.Binary, true,
+                            cancellationToken);
+                        } catch(OperationCanceledException ex)
+                        {
+                            _logger?.LogError("The WebSocket operation was cancelled");
+                        }
+                    }
+                }
+            }
+        }
+
+        private async Task Connect(Uri url)
+        {
+            var uriBuilder = new UriBuilder(url);
+            if (url.Scheme == "http")
+            {
+                uriBuilder.Scheme = "ws";
+            }
+            else if (url.Scheme == "https")
+            {
+                uriBuilder.Scheme = "wss";
+            }
+
+            await _webSocket.ConnectAsync(uriBuilder.Uri, _cancellationToken);
+        }
+
+        public void Dispose()
```

Comment:
Let's discuss this later today. 
(Also, I just realized that there is a race in case someone calls StartAsync on the connection without awaiting and Dispose the connection).